backend/voting/models/ranking.py

```diff
@@ -26,6 +27,11 @@ def __str__(self):
     def save(self, *args, **kwargs):
         super(RankRequest, self).save(*args, **kwargs)
 
+        exising_submissions = self.rank_submissions.all()
+
+        # do not recreate ranking
+        if exising_submissions:
+            return
```

Comment:
Uhm, this feels like it will fetch the objects just to check if it is empty or not?


```suggestion
        # do not recreate ranking
        if self.rank_submissions.exists():
            return
```

does this work? so it just checks with `SELECT EXISTS` and be more performant?